modules/common/cronjob/cronjob.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2022 Red Hat
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cronjob
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	batchv1 "k8s.io/api/batch/v1"
+	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+// NewCronJob returns an initialized CronJob.
+func NewCronJob(
+	cronjob *batchv1.CronJob,
+	timeout time.Duration,
+) *CronJob {
+	return &CronJob{
+		cronjob: cronjob,
+		timeout: timeout * time.Second,
```

Comment:
if timeout is a Duration, then we should not multiply it with time.Second here.

Reply:
Yes we have to use the raw Duration.